### PR TITLE
Fixes for compatibility with builds in firebase-cpp-sdk

### PR DESCRIPTION
**Pull request by wilhuff**
This makes it possible to embed the Firestore CMake build contained in this repo in the public C++ CMake build.

This is enough to get the firebase-cpp-sdk build working with tests disabled. More changes may be required for tests enabled.

### CODE REVIEW

File: cmake/compiler_setup.cmake [hunk 1]
```diff
@@ -116,14 +116,6 @@ if(MSVC)
   )
 endif()
 
-foreach(flag ${common_flags} ${c_flags})
```
Comment (var-const) [inline cmake/compiler_setup.cmake]: What's the reason to delete these lines?

File: cmake/external_rules.cmake [hunk 2]
```diff
@@ -36,9 +36,22 @@ function(download_external_sources)
...
 function(add_external_subdirectory NAME)
-  add_subdirectory(
-    ${FIREBASE_BINARY_DIR}/external/src/${NAME}
-    ${FIREBASE_BINARY_DIR}/external/src/${NAME}-build
-    EXCLUDE_FROM_ALL
-  )
+  string(TOUPPER ${NAME} UPPER_NAME)
+  if (NOT EXISTS ${${UPPER_NAME}_SOURCE_DIR})
+    set(${UPPER_NAME}_SOURCE_DIR "${FIREBASE_BINARY_DIR}/external/src/${NAME}")
+    set(${UPPER_NAME}_SOURCE_DIR "${${UPPER_NAME}_SOURCE_DIR}" PARENT_SCOPE)
```
Comment (var-const) [inline cmake/external_rules.cmake]: Can these two calls to `set` be rolled into one? (i.e. `set(${UPPER_NAME}_SOURCE_DIR "${FIREBASE_BINARY_DIR}/external/src/${NAME}" PARENT_SCOPE)`)

File: cmake/external_rules.cmake [hunk 3]
```diff
@@ -36,9 +36,22 @@ function(download_external_sources)
...
+  endif()
+
+  if (NOT EXISTS ${${UPPER_NAME}_BINARY_DIR})
+    set(${UPPER_NAME}_BINARY_DIR "${${UPPER_NAME}_SOURCE_DIR}-build")
+    set(${UPPER_NAME}_BINARY_DIR "${${UPPER_NAME}_BINARY_DIR}" PARENT_SCOPE)
+  endif()
+
+  if (EXISTS "${${UPPER_NAME}_SOURCE_DIR}/CMakeLists.txt")
```
Comment (var-const) [inline cmake/external_rules.cmake]: `else` branch does nothing -- it is an expected case as well, or should it report an error?

File: CMakeLists.txt [hunk 4]
```diff
@@ -187,7 +199,6 @@ else()
   )
 endif()
 
-add_alias(protobuf::libprotobuf libprotobuf)
```
Comment (var-const) [inline CMakeLists.txt]: What's the reason to remove this line?

File: scripts/binary_to_array.py [hunk 5]
```diff
@@ -48,8 +48,6 @@
 import logging
 import os
```
Comment (var-const) [inline scripts/binary_to_array.py]: What was the problem with the previous state of this script?

File: cmake/compiler_setup.cmake (hunk 1 above)
Comment (wilhuff) [reply cmake/compiler_setup.cmake]: These change global state and apply our preferences for flags to all compile actions. This breaks `firebase-cpp-sdk` and its dependencies which don't hold themselves to anywhere near as high a standard.

File: cmake/external_rules.cmake (hunk 2 above)
Comment (wilhuff) [reply cmake/external_rules.cmake]: No. Each `set` call can only operate on a single scope at a time.

File: CMakeLists.txt (hunk 4 above)
Comment (wilhuff) [reply CMakeLists.txt]: The upstream protobuf CMakeLists.txt now does this. Some previous upgrade failed to notice this and it's just cruft now.

File: scripts/binary_to_array.py (hunk 5 above)
Comment (wilhuff) [reply scripts/binary_to_array.py]: Most of the context for these changes was in the individual commit messages (e.g. [this one](https://github.com/firebase/firebase-ios-sdk/pull/4984/commits/07d1e5e1a72abb405e11d992b6f73789317ab4da)).

This code previously expected to be run from within a git repository but this assumption is flawed:

  * iOS users can download an archive of the source tree, in which case there's no `.git` folder
  * `firebase-cpp-sdk` actually does this to download its copy of `firebase-ios-sdk`

File: cmake/external_rules.cmake (hunk 3 above)
Comment (wilhuff) [reply cmake/external_rules.cmake]: Its an expected case that matches behavior in `firebase-cpp-sdk`. I may revisit this later.

**Pull request merged**